### PR TITLE
Delete the native implementation for ui.decodeImageFromList.

### PR DESCRIPTION
**Pull request by amirh**
The new ui.Codec API is a superset of the implementation in
image_decoding.cc. While we are not using ui.decodeImageFromList in the
framework anymore, we keep the dart function for backward compatability, and
just make it a thin wrapper around the codec API.

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Do we need to mark it for later removal in any way?  Or is it still useful in other contexts?  FYI @Hixie

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -928,9 +928,20 @@ Future<Codec> instantiateImageCodec(Uint8List list) {
...
   native "instantiateImageCodec";
 
-/// Convert an image file from a byte array into an [Image] object.
-void decodeImageFromList(Uint8List list, ImageDecoderCallback callback)
-    native "decodeImageFromList";
+/// Loads a single image frame from a byte array into an [Image] object.
+///
+/// This is a convenience wrapper around [instantiateImageCodec].
+/// Prefer using [instantiateImageCodec] which also supports multi frame images.
+// We only keep this method for backward compatibility.
```
Comment (Hixie) [inline lib/ui/painting.dart]: I would remove this comment. If you think about it, everything we support is something we support for backwards compatibility. :-)

**Comment by Hixie** [comment]
> Do we need to mark it for later removal in any way? 

Having declared API stability means we don't change the API unless we have to. Here, we don't have to.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**